app/grandchallenge/hanging_protocols/forms.py

```diff
@@ -107,15 +109,17 @@ def clean_json(self):
         ]
         for viewport in json:
             if "parent_id" in viewport:
-                errors.append(
-                    *self._validate_parent_id(viewport, viewport_names)
-                )
+                errors = [
+                    *errors,
+                    *self._validate_parent_id(viewport, viewport_names),
+                ]

```

Comment:
Nit: cleaner to use `errors.extend(self._validate_parent_id(viewport, viewport_names))` etc.

Reply:
Or rather than creating, returning and appending errors, can't you call `self.add_error(error=error, field="json")` each time?